### PR TITLE
[FEATURE] Update usage stats schema to account for RBP's run() payload

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Update JSON schema to account for new payload corresponding to `RBP.run()`
- Tests to validate schema changes

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/core/usage_statistics/schemas.py [hunk 1]
```diff
@@ -750,6 +750,135 @@
     "additionalProperties": False,
 }
 
+anonymized_domain_builder_schema = {
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "anonymized-domain-builder",
+    "definitions": {
+        "anonymized_string": anonymized_string_schema,
+        "anonymized_batch_request": anonymized_batch_request_schema,
+        "anonymized_datasource_name": anonymized_datasource_name_schema,
```
Comment (cdkini) [inline great_expectations/core/usage_statistics/schemas.py:759]: No such luck narrowing this down @anthonyburdi @alexsherstinsky - if the definition is used anywhere in the hierarchy of the given object, it must be included in the definitions keys.

I'll keep reading the JSON Schema guide to see if we have a better solution but this seems to track with the rest of the schemas noted herein.

Comment (anthonyburdi) [reply great_expectations/core/usage_statistics/schemas.py:759]: All good, thanks for checking!

**Comment by alexsherstinsky** [comment]
@cdkini Everything looks good -- I just think that the reason that the usage statistics messages tests are failing is because they run against the QA environment.  So for that you actually have to do the following:
1) Get the schema into DEV environment;
2) Implement the messages tests and run them (be sure to turn off `aws` by commenting out the decorator `@pytest.mark.aws_integration` in `tests/integration/usage_statistics/test_usage_statistics_messages.py`) and setting the URL to `dev` (in `tests/integration/usage_statistics/test_integration_usage_statistics.py` on line 13);
3) Issue pull request from the internal repository with the JSON schema update;
4) Once all tests pass in `dev`, deploy to `QA` -- and now this PR should get all tests passing -- you may want to ask @anthonyburdi and me to re-review this PR when the messages tests are in it (just in case).
5) Once this PR is merged, deploy the JSON schema to the `PROD` environment.

@anthonyburdi Did I miss anything or give incorrect pointers?  Thanks!

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
@cdkini Please see comments -- getting close -- happy to discuss.  Thanks!

**Comment by anthonyburdi** [comment]
> @cdkini Everything looks good -- I just think that the reason that the usage statistics messages tests are failing is because they run against the QA environment. So for that you actually have to do the following:
> 
> 1. Get the schema into DEV environment;
> 2. Implement the messages tests and run them (be sure to turn off `aws` by commenting out the decorator `@pytest.mark.aws_integration` in `tests/integration/usage_statistics/test_usage_statistics_messages.py`) and setting the URL to `dev` (in `tests/integration/usage_statistics/test_integration_usage_statistics.py` on line 13);
> 3. Issue pull request from the internal repository with the JSON schema update;
> 4. Once all tests pass in `dev`, deploy to `QA` -- and now this PR should get all tests passing -- you may want to ask @anthonyburdi and me to re-review this PR when the messages tests are in it (just in case).
> 5. Once this PR is merged, deploy the JSON schema to the `PROD` environment.
> 
> @anthonyburdi Did I miss anything or give incorrect pointers? Thanks!

@alexsherstinsky You said it perfectly! We just did step 4, ready to merge this PR!

File: tests/integration/usage_statistics/test_usage_statistics_messages.py [hunk 2]
```diff
@@ -2497,6 +2497,75 @@ def test_generate_messages_with_defaults():
...
+            "event_payload": {},
+            "success": True,
+            "version": "1.0.0",
+            "event_time": "2020-06-25T16:08:28.070Z",
+            "data_context_id": "00000000-0000-0000-0000-000000000002",
+            "data_context_instance_id": "10000000-0000-0000-0000-000000000002",
+            "ge_version": "0.14.7.manual_testing",
+            "x-forwarded-for": "00.000.00.000, 00.000.000.000",
+        },
+        {
```
Comment (anthonyburdi) [inline tests/integration/usage_statistics/test_usage_statistics_messages.py:2512]: I would like to see one more example event message here with custom classes.

**Review by anthonyburdi — APPROVED** [review]
Approving to not slow you down but I would like to see one more example event using custom classes. Otherwise all looks great, love the handling of "required" fields.

File: tests/integration/usage_statistics/test_usage_statistics_messages.py [hunk 3]
```diff
@@ -2497,6 +2497,136 @@ def test_generate_messages_with_defaults():
...
+            "event": "profiler.run",
+            "event_payload": {
+                "anonymized_name": "5b6c98e19e21e77191fb071bb9e80070",
+                "anonymized_rules": [
```
Comment (cdkini) [inline tests/integration/usage_statistics/test_usage_statistics_messages.py:2572]: @anthonyburdi here's the additional test case with 2 rules. The first has recognized parents and the second does not.

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**